src/client.cpp

```diff
@@ -324,26 +323,17 @@ ADDON_STATUS ADDON_SetSetting(const char *settingName, const void *settingValue)
   }
   else if (str == "reseticons")
   {
-    if ( g_eraseIcons != *(bool*)settingValue )
+    if (*(bool*)settingValue == true)
```

Comment:
C++ style cast

Reply:
Fixed with move to Settings.cpp templates

Reply:
Old Code